src/core/device-interface.h

```diff
@@ -30,6 +32,7 @@ class stream_profile_interface;
 //
 class device_interface
     : public virtual info_interface
+    , public features_container
```

Comment:
I can see how the `device_interface` is a `supports_features_interface`, but why the container? Not every device will support features...

Reply:
Devices that don't register any features will have an empty list of features. 
Some logic access the device as a `device_interface` (getting sensor owner for example) I don't want to dynamically cast it to `device` for each use

Reply:
So they should have a default implementation of supports/get feature, but they don't need a map of features and other details...
The device **interface** should not assume an implementation.